pkg/operator/ceph/object/zone/controller.go

```diff
@@ -396,3 +425,33 @@ func (r *ReconcileObjectZone) updateStatus(observedGeneration int64, name types.
 	}
 	logger.Debugf("object zone %q status updated to %q", name, status)
 }
+
+func (r *ReconcileObjectZone) deletePoolsAndZone(objContext *object.Context, zone *cephv1.CephObjectZone, realmName string) error {
+	if !zone.Spec.PreservePoolsOnDelete {
+		realmArg := fmt.Sprintf("--rgw-realm=%s", realmName)
+		zoneGroupArg := fmt.Sprintf("--rgw-zonegroup=%s", zone.Spec.ZoneGroup)
+		zoneArg := fmt.Sprintf("--rgw-zone=%s", zone.Name)
+
+		accessKeyArg, secretKeyArg, err := object.GetRealmKeyArgs(r.opManagerContext, r.context, realmName, zone.Namespace)
+		if err != nil {
+			return errors.Wrap(err, "failed to get keys for realm")
+		}
+		args := []string{"zone", "delete", realmArg, zoneGroupArg, zoneArg, accessKeyArg, secretKeyArg}
+
+		output, err := object.RunAdminCommandNoMultisite(objContext, false, args...)
+		if err != nil {
+			return errors.Wrapf(err, "failed to create ceph zone %q for reason %q", zone.Name, output)
+		}
+
+		logger.Debugf("deleting pools for ceph zone %q", zone.Name)
+
+		err = object.DeletePools(objContext, zone.Spec.DataPool, zone.Spec.MetadataPool, false)
```

Comment:
instead of passing in pool names like `zone.Spec.DataPool, zone.Spec.MetadataPool`, consider using `radosgw-admin zone get` to list all the real pool names

Reply:
Yaa that's valid for a single site to multisite conversion use case.